mkt/developers/tasks.py

```diff
@@ -338,17 +339,34 @@ def save_icon(obj, icon_content):
     dirname = obj.get_icon_dir()
     destination = os.path.join(dirname, '%s' % obj.pk)
     remove_icons(destination)
-    resize_icon(tmp_dst, destination, mkt.CONTENT_ICON_SIZES,
-                set_modified_on=[obj])
+    icon_hash = resize_icon(tmp_dst, destination, mkt.CONTENT_ICON_SIZES,
+                            set_modified_on=[obj])
 
-    # Need to set the icon type so .get_icon_url() works
-    # normally submit step 4 does it through AppFormMedia,
-    # but we want to beat them to the punch.
-    # resize_icon outputs pngs, so we know it's 'image/png'
+    # Need to set icon type so .get_icon_url() works normally
+    # submit step 4 does it through AppFormMedia, but we want to beat them to
+    # the punch. resize_icon outputs pngs so we know it's 'image/png'.
+    obj.icon_hash = icon_hash['icon_hash']  # In case, we're running not async.
```

Comment:
This gave me pause because I wondered if we could end up caching the un-optimised images on the CDN with the final URL. But `pngcrush_image()` does change the hash, so we should be fine.